packages/neuron-ui/src/components/HDWalletSign/index.tsx

```diff
@@ -56,21 +56,11 @@ const HDWalletSign = ({ tx }: { tx: State.DetailedTransaction }) => {
         } else {
           try {
             if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
-              address = ckbCore.utils.bech32Address(cell.lock.args, {
-                prefix: addressPrefix,
-                type: ckbCore.utils.AddressType.HashIdx,
-                codeHashOrCodeHashIndex: '0x00',
-              })
+              address = generateAddress(cell.lock, { config })
+            } else if (['data', 'type', 'data1', 'data2'].includes(cell.lock.hashType)) {
```

Comment:
Is there a reason for validating the hashType instead of directly validating encodeToAddress(cell.lock, { config })? 

Note that encodeToAddress will check if a script is valid.

Reply:
It's based on an earlier conversation.
<img width="798" alt="image" src="https://github.com/nervosnetwork/neuron/assets/7459812/e8f3ba34-7615-49ae-a33c-bc7d3eeb48a3">


This is definitely simpler
```ts
if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
  address = generateAddress(cell.lock, { config })
} else {
  address = encodeToAddress(cell.lock, { config })
}
```
Erorr (if any)  thrown
```ts
export function ValidateScript(
  ...
  throw new Error(`${debugPath}.hashType must be either data or type!`);
```

The current code is less simple but has more detailed error message.
```ts
throw new Error(`Invalid lock script hash type ${cell.lock.hashType}`)
```

I think they both work, so we just need to pick one. @homura

Reply:
Pushed changes. Simple is better.
```ts
if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
  address = generateAddress(cell.lock, { config })
} else {
  address = encodeToAddress(cell.lock, { config })
}
```

Reply:
cool, it looks fine